interfaces/builtin/hardware_control.go

```diff
@@ -0,0 +1,51 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const hardwareControlSummary = `allows control of system hardware`
+
+const hardwareControlBaseDeclarationSlots = `
+  hardware-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const hardwareControlConnectedPlugAppArmor = `
+# Description: This interface allows for controlling hardware and platform
+# devices connected to the system.
+# This is reserved because it allows potentially disruptive operations and
+# access to devices which may contain sensitive information.
+
+# files in /sys pertaining to hardware (eg, 'lspci -A linux-sysfs')
```

Comment:
Do you need any access to device nodes (character and block devices)? IIRC sysfs doesn't expose them directly.

Reply:
You are correct about the device nodes - I wanted to be careful with the wording here so it was clear that this was an interface which could be used to disrupt a system.

I also am not aware of a direct way /sys could grant direct access to block contents or provide direct access to a peripheral. I mostly wanted to be explicit about the capabilities for disruption here, things like ejecting block devices, deleting device nodes, changing device parameters. I'm happy to amend this to be a little less alarmist.

Reply:
I was thinking about the intended use case (to make sure it is captured), not about security (the implications of granting such permission).